open-grant-proposals/open-proposal-twitter-thread-rollup-backed-by-filecoin-and-nfts.md

```diff
@@ -0,0 +1,127 @@
+
+- Existing bots/solutions traditionally rely on centralised (hidden) storage and therefore limited in the what can be done with the thread that is snapshot'd.
+
+- The creation of a NFT derived from the timestamped twitter thread combined with a Filecoin stored rollup will bring two new things to the party:
+    - Facilitating trading of the thread - popular, viral or controversial threads may gain in value 
+    - Creating a source of revenue to maintain the tweeter bot and its Filecoin storage costs - we could possible charge $1-2 for creating the NFT as well as enabling trading on OpenSea with a secondary sales commission of 2% which will then go back into maintain the Filecoin storage
+    - Popularising Filecoin in general on NFT platforms - often NFTs use IPFS but in theory should be migrating to a 
+    Filecoin solution with a proper incentivisation layer - this would be a perfect candidate to test this out theory and popularise its usage
+
+## Value
+
+- What are the benefits to getting this right?
+
+* An example of how to use Filecoin in an NFT (popular future usecase) and an open source Github demo project
+* Promote the use of Filecoin in the twitter sphere in general
+* Provide a simple toolset that can be forked and used by many and demonstrates Filecoin
+
+- What are the risks if you don't get it right?
+
+* There may be reputational risk for the project if it fails, rendering the twitter bot unloved or sidelined.  
+
+- What are the risks that will make executing on this project difficult?
+
+I believe the technologies is not the hard part at least for ourselves, we have good experience in IPFS, building webapps and NFTs however there may be a problem in the incentivisation model in that supporting the file storage may become costly overtime. One option to avoid this is to snapshot the thread and store centrally, only pushing to filecoin when an NFT is created and the small fee is paid. 
+ 
+Filecoin is still emerging and the APIs may change in the short/medium term - keeping up to date may become probalatic but after review we think it is a fairly simply setup so this shouldn't be seen a huge risk right now.
+
+## Deliverables
+
+The deliverable for the project will be as follows:
+
+- Twitter bot - hosted on GCP/AWS (free tier most likely is suitable)
+
+- Filecoin integration to capture the thread and store on Filecoin
+
+- Simple one page web based UI to see rolled up threads and buy them as an NFT is possible
+    - Hosted on a free service like Firebase
+    - Create a simple DB for stored some metadata about the thread
+    - Simple minting functionality to create the NFT with web3 integration
+     
+- NFT contract that allows user to mint and pay small nominal fee (this can easily by added to OpenSea for trading)
+    - The NFT `tokenURI()` will utilise the Filecoin hosted image
+
```

Comment:
If this can also show or link to proof that the image is being stored on Filecoin and for how long that would be great. 

You could also explore how pending expiration of the storage deal affects users/the market for super rare items.

Reply:
Ye good idea, the longevity of the image may actually become a value attribute of the NFT itself.